File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
         String str="";

        for (int i = 0; i < s.length(); i++) {
            if (Character.isLetterOrDigit(s.charAt(i))) {
                str += Character.toLowerCase(s.charAt(i));
            }
        }
        System.out.println(str);
        return isValid(str.toLowerCase());
    }
    static boolean isValid(String s){
        for (int i = 0; i < s.length()/2; i++) {
            if (s.charAt(i)!=s.charAt(s.length()-i-1)){
                return false;
            }
        }
        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        t = "".join(char for char in s if char.isalnum()).lower()
        i=0
        j=len(t)-1
        while j>i:
            if t[j]!=t[i]:
                return False
            j-=1
            i+=1
        
        return True